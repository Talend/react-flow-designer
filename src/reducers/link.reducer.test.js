import { Map } from 'immutable';

import linkReducer from './link.reducer';
import { LinkRecord, PortRecord } from '../constants/flowdesigner.model';

describe('check linkreducer', () => {
	const initialState = new Map()
		.set('links', new Map()
			.set('id1', new LinkRecord({
				id: 'id1',
			}))
		).set('ports', new Map()
			.set('id1', new PortRecord({
				id: 'id1',
			}))
		);

	it('FLOWDESIGNER_LINK_REMOVE should remove link from state', () => {
		expect(linkReducer(initialState, { type: 'FLOWDESIGNER_LINK_REMOVE', linkId: 'id1' }))
		.toEqual(new Map()
		.set('links', new Map())
		.set('ports', new Map()
			.set('id1', new PortRecord({
				id: 'id1',
			}))
		));
	});


	it('FLOWDESIGNER_LINK_SET_TARGET switch target to correct port if it exist', () => {
		expect(linkReducer(initialState,
			{ type: 'FLOWDESIGNER_LINK_SET_TARGET', linkId: 'id1', targetId: 'id1' }
		)).toEqual(new Map()
		.set('links', new Map()
			.set('id1', new LinkRecord({
				id: 'id1',
				targetId: 'id1',
			}))
		).set('ports', new Map()
			.set('id1', new PortRecord({
				id: 'id1',
			}))
		));
	});

	it('FLOWDESIGNER_LINK_SET_SOURCE switch source to correct port if it exist', () => {
		expect(linkReducer(initialState,
			{ type: 'FLOWDESIGNER_LINK_SET_SOURCE', linkId: 'id1', sourceId: 'id1' }
		)).toEqual(new Map()
		.set('links', new Map()
			.set('id1', new LinkRecord({
				id: 'id1',
				sourceId: 'id1',
			}))
		).set('ports', new Map()
			.set('id1', new PortRecord({
				id: 'id1',
			}))
		));
	});
});
