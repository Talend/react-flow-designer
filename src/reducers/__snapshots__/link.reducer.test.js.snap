exports[`check linkreducer FLOWDESIGNER_LINK_ADD should add a new link to the state 1`] = `
Object {
  "in": Object {
    "id2": Object {
      "id2": Object {
        "id2": "id2",
      },
    },
  },
  "links": Object {
    "id1": Object {
      "attributes": Object {
        "attr": "attr",
      },
      "id": "id1",
      "linkType": undefined,
      "sourceId": "id1",
      "targetId": "id2",
    },
    "id2": Object {
      "attributes": Object {},
      "id": "id2",
      "linkType": undefined,
      "sourceId": "id1",
      "targetId": "id2",
    },
  },
  "nodes": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
  },
  "out": Object {
    "id1": Object {
      "id1": Object {
        "id2": "id2",
      },
    },
  },
  "ports": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id3": Object {
      "attributes": undefined,
      "id": "id3",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id4": Object {
      "attributes": undefined,
      "id": "id4",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
  },
  "preds": Object {
    "id2": Object {
      "id1": "id1",
    },
  },
  "sucs": Object {
    "id1": Object {
      "id2": "id2",
    },
  },
}
`;

exports[`check linkreducer FLOWDESIGNER_LINK_REMOVE should remove link from state 1`] = `
Object {
  "links": Object {},
  "nodes": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
  },
  "ports": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id3": Object {
      "attributes": undefined,
      "id": "id3",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id4": Object {
      "attributes": undefined,
      "id": "id4",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
  },
}
`;

exports[`check linkreducer FLOWDESIGNER_LINK_REMOVE_ATTR should remove a specific attributes from attr map 1`] = `
Object {
  "links": Object {
    "id1": Object {
      "attributes": Object {},
      "id": "id1",
      "linkType": undefined,
      "sourceId": "id1",
      "targetId": "id2",
    },
  },
  "nodes": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
  },
  "ports": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id3": Object {
      "attributes": undefined,
      "id": "id3",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id4": Object {
      "attributes": undefined,
      "id": "id4",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
  },
}
`;

exports[`check linkreducer FLOWDESIGNER_LINK_SET_ATTR should merge attributes within link attr property 1`] = `
Object {
  "links": Object {
    "id1": Object {
      "attributes": Object {
        "attr": "attr",
        "selected": false,
      },
      "id": "id1",
      "linkType": undefined,
      "sourceId": "id1",
      "targetId": "id2",
    },
  },
  "nodes": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
  },
  "ports": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id3": Object {
      "attributes": undefined,
      "id": "id3",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id4": Object {
      "attributes": undefined,
      "id": "id4",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
  },
}
`;

exports[`check linkreducer FLOWDESIGNER_LINK_SET_SOURCE switch source to correct port if it exist 1`] = `
Object {
  "links": Object {
    "id1": Object {
      "attributes": Object {
        "attr": "attr",
      },
      "id": "id1",
      "linkType": undefined,
      "sourceId": "id4",
      "targetId": "id2",
    },
  },
  "nodes": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
  },
  "out": Object {
    "id1": Object {
      "id4": Object {
        "id1": "id1",
      },
    },
  },
  "ports": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id3": Object {
      "attributes": undefined,
      "id": "id3",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id4": Object {
      "attributes": undefined,
      "id": "id4",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
  },
}
`;

exports[`check linkreducer FLOWDESIGNER_LINK_SET_TARGET switch target to correct port if it exist 1`] = `
Object {
  "in": Object {
    "id2": Object {
      "id3": Object {
        "id1": "id1",
      },
    },
  },
  "links": Object {
    "id1": Object {
      "attributes": Object {
        "attr": "attr",
      },
      "id": "id1",
      "linkType": undefined,
      "sourceId": "id1",
      "targetId": "id3",
    },
  },
  "nodes": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeSize": undefined,
      "nodeType": undefined,
      "position": undefined,
    },
  },
  "ports": Object {
    "id1": Object {
      "attributes": undefined,
      "id": "id1",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
    "id2": Object {
      "attributes": undefined,
      "id": "id2",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id3": Object {
      "attributes": undefined,
      "id": "id3",
      "nodeId": "id2",
      "portType": undefined,
      "position": undefined,
    },
    "id4": Object {
      "attributes": undefined,
      "id": "id4",
      "nodeId": "id1",
      "portType": undefined,
      "position": undefined,
    },
  },
}
`;
